lib/components/DSButton.js

```diff
@@ -1,46 +1,24 @@
 import React from 'react';
 import '../../lib/styles/styles.scss';
 
-let type = '';
-let size = '';
+const getClassNameFromProp = (prop, prefix) => (prop && `${prefix}${prop}`) || '';
+const getSizeClassName = size => getClassNameFromProp(size);
+const getTypeClassName = type => getClassNameFromProp(type);
 
-const checkType = (props) => {
-  if (props.type === 'primary') {
-    type = "ds-btn-primary";
-  } else if (props.type === 'secondary') {
-    type = "ds-btn-secondary";
-  } else if (props.type === 'success') {
-    type = "ds-btn-success";
-  } else if (props.type === 'danger') {
-    type = "ds-btn-danger";
-  } else if (props.type === 'info') {
-    type = "ds-btn-info";
-  } else if (props.type === 'ghost') {
-    type = "ds-btn-ghost";
-  } else {
-    type = "ds-btn-primary";
-  }
-  return type;
+const DSButton = ({ type, circle, icon, size, children, ...rest }) => {
+  const prefix = circle ? 'ds-circle-btn-' : 'ds-btn-';
+  return circle ? (
+    <button className={`ds-circle-btn ${getTypeClassName(type, prefix)}`} {...rest}>
+      {icon ? <i className="material-icons ds-icon-small">{props.icon}</i> : { children }}
```

Comment:
Is {`children`} the same thing as transclusion in Angular? I am assuming so..

Reply:
yeah - generally to leave things open for extension you don't want to restrict the input (i.e. if you used a text prop, the child could only be text, but if you allow children you leave it open